python/ray/tune/trainable.py

```diff
@@ -339,11 +335,20 @@ def restore_from_object(self, obj):
 
         info = pickle.loads(obj)
         data = info["data"]
-        tmpdir = tempfile.mkdtemp("restore_from_object", dir=self.logdir)
-        checkpoint_path = os.path.join(tmpdir, info["checkpoint_name"])
+        tmpdir = info["savedir"]
+        if not os.path.exists(tmpdir):
+            # We recreate the same subtree and path.
+            os.makedirs(tmpdir)
```

Comment:
Why change this logic? This will try to recreate the absolute path right? Whereas, the previous code recreated it relative to the given tempdir.

Reply:
So this was one way to enable `tf.Saver` functionality. 
```
saver = tf.Saver() 
prefix = self.saver.save(self.sess, checkpoint_dir + "/save", global_step=self.iterations)
```

`tf.Saver` saves things into a given directory with this format:
```shell
╰─ ls  /Users/rliaw/ray_results/mnist_hyperband_test/TrainMNIST_0_learning_rate=0.00022789_2019-05-01_15-21-20lzl3rbbe/tmpudqrk34zsave_to_object/checkpoint_2/
checkpoint                 save-2.data-00000-of-00001 save-2.index               save-2.meta                save-2.tune_metadata
```

And, in `checkpoint`, it saves absolute paths:
```shell
╰─ cat  /Users/rliaw/ray_results/mnist_hyperband_test/TrainMNIST_0_learning_rate=0.00022789_2019-05-01_15-21-20lzl3rbbe/tmpudqrk34zsave_to_object/checkpoint_2/checkpoint
model_checkpoint_path: "/Users/rliaw/ray_results/mnist_hyperband_test/TrainMNIST_0_learning_rate=0.00022789_2019-05-01_15-21-20lzl3rbbe/tmpudqrk34zsave_to_object/checkpoint_2/save-2"
all_model_checkpoint_paths: "/Users/rliaw/ray_results/mnist_hyperband_test/TrainMNIST_0_learning_rate=0.00022789_2019-05-01_15-21-20lzl3rbbe/tmpudqrk34zsave_to_object/checkpoint_2/save-2"
```

There's alternative workarounds for this problem (i.e., doing a os.chdir in `trainable:save_to_object()`), but this seemed reasonable...

Reply:
Won't this break when I move it to some other machine which does have /Users/rliaw?